okio/src/main/java/okio/Buffer.java

```diff
@@ -86,6 +90,45 @@ public long size() {
     };
   }
 
+  @Override public Writer writer() {
+    return writer(Util.UTF_8);
+  }
+
+  @Override public Writer writer(final Charset charset) {
+    if (charset == null) throw new NullPointerException("charset == null");
+    return new Writer() {
+      @Override public void write(char[] cbuf, int off, int len) throws IOException {
+        writeChars(charset, cbuf, off, len);
+      }
+
+      @Override public void flush() throws IOException {
+      }
+
+      @Override public void close() throws IOException {
+      }
+
+      @Override public String toString() {
+        return Buffer.this + ".writer(" + charset.name() + ")";
+      }
+    };
+  }
+
+  void writeChars(Charset charset, char[] cbuf, int off, int len) {
```

Comment:
(this could happen if say, you’re chunking chars without paying attention to their boundaries)